internal/service/server/server.go

```diff
@@ -1214,7 +1226,31 @@ func (r *serverResource) Update(ctx context.Context, req resource.UpdateRequest,
 		return
 	}
 
-	if changeRequiresServerStop(state, plan) {
+	// Check if plan is being changed and handle hot resize if enabled
+	if !plan.Plan.Equal(state.Plan) && plan.HotResize.ValueBool() {
+		// Attempt hot resize
+		hotResizeReq := &request.ModifyServerRequest{
+			UUID: uuid,
+		}
+
+		// Set the appropriate fields based on whether it's a custom plan
+		if plan.Plan.ValueString() == customPlanName {
+			hotResizeReq.CoreNumber = int(plan.CPU.ValueInt64())
+			hotResizeReq.MemoryAmount = int(plan.Mem.ValueInt64())
+		} else {
+			hotResizeReq.Plan = plan.Plan.ValueString()
+		}
+
+		_, err := r.client.ModifyServer(ctx, hotResizeReq)
+		if err != nil {
+			// If hot resize fails, return an error
+			resp.Diagnostics.AddError(
+				"Hot resize failed",
+				fmt.Sprintf("Unable to hot resize server: %s. Hot resize requires the server to be in a state where it can be resized without a reboot. If hot resize is not possible, set hot_resize = false to use the standard approach with server reboot.", utils.ErrorDiagnosticDetail(err)),
+			)
+			return
+		}
+	} else if changeRequiresServerStop(state, plan) {
```

Comment:
Can this `else if` cause problems if same plan contains hot-resize change and changes that require server restart 🤔

Reply:
Yeah I agree, might become an issue. I think we could change the order? Pseudo code:

```
if needsServerStop {
   // ...
} else if planChanged && hotResizeEnabled {
   // ...
}
```

Reply:
I built a test case around this. Creates a server with a plan + hot resizing, once applied it adds a utility network but also changes the plan. The test fails because:

```
        Error: Invalid server state

          with upcloud_server.mixed_changes,
          on terraform_plugin_test.tf line 12, in resource "upcloud_server" "mixed_changes":
          12:                                   resource "upcloud_server" "mixed_changes" {

        Server needs to be stopped to alter networks
```

I'll add a fix for this, thanks Toni

Reply:
Fix in https://github.com/UpCloudLtd/terraform-provider-upcloud/pull/747/commits/fc40daa1372fab7717a0e80335ab60c12a45ecf6. Ran the tests locally and seems to be all fine, just takes 3-5 minutes to finish.

Reply:
Yeah, the reverse ordering should be safer 👌